Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -437,8 +437,16 @@ static void populateStencilOperation(MTLStencilDescriptor *mtlStencil, const WGP
     if (descriptor.vertex.bufferCount)
         mtlRenderPipelineDescriptor.vertexDescriptor = createVertexDescriptor(descriptor.vertex);
 
-    if (descriptor.primitive.nextInChain)
-        return RenderPipeline::createInvalid(*this);
+    MTLDepthClipMode mtlDepthClipMode = MTLDepthClipModeClip;
+    if (descriptor.primitive.nextInChain) {
+        if (descriptor.primitive.nextInChain->sType != WGPUSType_PrimitiveDepthClipControl)
+            return RenderPipeline::createInvalid(*this);
+
+        auto* depthClipControl = reinterpret_cast<const WGPUPrimitiveDepthClipControl*>(descriptor.primitive.nextInChain);
+
+        if (depthClipControl->unclippedDepth)
+            mtlDepthClipMode = MTLDepthClipModeClamp;
```

Comment:
We don't have to do any workarounds or anything?

Reply:
The extent of the workaround is the WGSL compiler should clamp fragDepth, when written, to the viewport z-bounds. I will file a bug for this

Reply:
https://bugs.webkit.org/show_bug.cgi?id=252729